src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedItems = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1 && this[i] === undefined) {
+        break;
+      }
+
+      if (this[i] === undefined || this[i] === null) {
+        joinedItems += separator;
+        continue;
+      }
+
+      if (i === this.length - 1) {
+        joinedItems += this[i];
+        break;
+      }
+
+      joinedItems += this[i] + '' + separator;
```

Comment:
```suggestion
      joinedItems += this[i] + separator;
```

do you really need this `+ ''`?

Reply:
I've made it to complete the test:  if the separator is `Null`.